Index: src/main/java/org/codehaus/mojo/wagon/SshExecMojo.java
===================================================================
--- src/main/java/org/codehaus/mojo/wagon/SshExecMojo.java	(revision 0)
+++ src/main/java/org/codehaus/mojo/wagon/SshExecMojo.java	(working copy)
@@ -0,0 +1,38 @@
+package org.codehaus.mojo.wagon;
+
+import java.io.IOException;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.wagon.CommandExecutor;
+import org.apache.maven.wagon.Wagon;
+import org.apache.maven.wagon.WagonException;
+
+/**
+ * Executes a list of commands against a given server.
+ * 
+ * @goal sshexec
+ * @requiresProject false
+ */
+public class SshExecMojo extends AbstractSingleWagonMojo
+{
+
+    /**
+     * The commands that we will execute.
+     *
+     * @parameter alias="commands"
+     * @required
+     */
+    private String[] commands;
+    
+    protected void execute(final Wagon wagon) throws MojoExecutionException, WagonException, IOException {
+        try {
+            if (commands != null) {
+                for (int i = 0; i < commands.length; i++) {
+                    ((CommandExecutor) wagon).executeCommand(commands[i]);
+                }
+            }
+        } catch (final WagonException e) {
+            this.getLog().error(e);
+        }
+    }
+}
